datafusion/sql/src/unparser/rewrite.rs

```diff
@@ -100,6 +100,79 @@ fn rewrite_sort_expr_for_union(exprs: Vec<SortExpr>) -> Result<Vec<SortExpr>> {
     Ok(sort_exprs)
 }
 
+/// Rewrite Filter plans that have a Window as their input by inserting a SubqueryAlias.
+///
+/// When a Filter directly operates on a Window plan, it can cause issues during SQL unparsing
+/// because window functions in a WHERE clause are not valid SQL. The solution is to wrap
+/// the Window plan in a SubqueryAlias, effectively creating a derived table.
+///
+/// Example transformation:
+///
+/// Filter: condition
+///   Window: window_function
+///     TableScan: table
+///
+/// becomes:
+///
+/// Filter: condition
+///   SubqueryAlias: __qualify_subquery
+///     Projection: table.column1, table.column2
+///       Window: window_function
+///         TableScan: table
+///
+pub(super) fn rewrite_qualify(plan: &LogicalPlan) -> Result<LogicalPlan> {
+    let plan = plan.clone();
```

Comment:
```suggestion
pub(super) fn rewrite_qualify(plan: LogicalPlan) -> Result<LogicalPlan> {
```

If we clone a reference it's better to make it known in the signature that we need ownership anyway (and do clone at callsite if required)